### ISSUE TITLE
Test and deploy Amazon Linux 2 Platform Upgrades (for PHP 7.4 upgrade) to the CMS Infrastructure

### ISSUE THREAD

**Issue by olivereri** [department-of-veterans-affairs/va.gov-cms#7777]
## Description
Changes were made to the build portion of shared infrastructure code.  Before merging our changes so that CMS infrastructure runs on Amazon Linux 2 with PHP 7.4 we must verify the build doesn't break other App's builds. Then once this is tested fully, we want to deploy this out to the CMS infrastructure (Production).

## Acceptance Criteria
- [x] Modified BRD Build process modifications don't break other App's builds. 
- [x] Branch containing Amazon Linux 2 and PHP 7.4 modifications merged into Devops default branch.
- [x] The updated AL2/PHP7.4 infra is merged and deployed to CMS Infrastructure (will be live in Prod)
  - [x] Roll back plan completed.

## Implementation Details 
* CMS can be configured to use the base AL1 server image and validate modifications don't affect building that image.
* Final deployment should happen after daily Prod deploy.

## Relations
#7583

**Comment by olivereri** [comment]
## Draft Explanation of shared BRD file chanages:

Shared files changes:

`ansible/build/create.yml`
Add conditional for overriding ansible_distribution_major_version set to 7. AL2 has a newer major version of the OS.


`ansible/build/roles/common/base-os/tasks/amazon.yml `
Install PIP for AL2.
Move Python selinux bindings to BEFORE the PIP install if using AL2. 
This will install the bindings BEFORE requirements file copied for AL1. Tested, no errors.
Add conditional to Python3 install task for AL1 only. Python3 is already installed AL2.

`ansible/deployment/config/jenkins-vetsgov/seed_job.groovy`
Add encrypted-base Jenkins job to create an encrypted base AMI using AL2. Adds parameter for specific AMI ID if needed.

`ansible/Jenkinsfiles/builds/base-build-al2`
Copy of encrypted-base Groovy script to create an encrypted base AMI using AL2. Adds parameter for specific AMI ID if needed.

`ansible/deployment/create.yml`
Add ansible task to wait for EC2 boot scripts (user-data) to finish.

`ansible/deployment/roles/deploy-launch-template/tasks/cloud-init-wait.yml`
Ansible task that waits for EC2 boot scripts (user-data) to finish before starting other tasks. On AL2 user-data was running at the same time as Ansible tasks causing failures.

**Comment by olivereri** [comment]
Minor concern that updating post deployment tasks might break PIV login for Prod.

**Comment by olivereri** [comment]
## Roll Back Plan

**If the Prod deployment fails during post-deployment, this is the state the deployment will be in.**
1. Prod shields will be up: A static PHP page that prevents any interaction with Drupal.
2. Prod previous site alert will still be active, indicating that a deployment is pending or in progress.
3. New Prod EC2 Instance will be `pending:wait` in the Auto Scaling Group. This can be cleaned up manually

**If, after deployment, Production Drupal **_is down_**  this is the state the deployment will be in.**

1. New EC2 Instance based on Amazon Linux is `inService`. Target Group will eventually signal that the new instance is failed in roughly 10 minutes. The ASG will terminate it then attempt to start a new instance. It will be stuck in `pending:wait` .
1. Revert [PR](https://github.com/department-of-veterans-affairs/devops/pull/10687)
2. Redeploy [Prod Job](http://jenkins.vfs.va.gov/job/deploys/job/cms-vagov-prod/) manually, using `b82ff38ca3a0f0e77aabc341e2b407050c30ffe4` as the ref parameter. This is the last commit to have an AMI based off of Amazon Linux 1.
2. [Rebuild ](http://jenkins.vfs.va.gov/job/builds/job/cms/) using the last passing commit of [va.gov-cms repository](https://github.com/department-of-veterans-affairs/va.gov-cms/commits). 20 Minutes worst case.
3. Redeploy [Prod Job](http://jenkins.vfs.va.gov/job/deploys/job/cms-vagov-prod/) manually, using the last passing commit of [va.gov-cms repository](https://github.com/department-of-veterans-affairs/va.gov-cms/commits).

deployment_version tag = latest
change to previous

**Comment by olivereri** [comment]
Follow-on work:

Some portions of the Build process that install binaries assumed Upstart as the init daemon. AL2 uses SystemD as its init daemon and requires different app service configuration. Some of the work required to update the install of binaries was completed in [[cms] Fix or add node_exporter and Promtail systemd service files #10702](https://app.zenhub.com/workspace/o/department-of-veterans-affairs/devops/issues/10702).  

There are still sections of configuration code that use old style Upstart init daemon commands that cause errors in the build process, and during the deploy process. Will need to go through the build and deploy code to add support for SystemD commands and unit files for installed binaries. This will help quiet errors and avoid problems like in #10702
